### ISSUE TITLE
Compiler error when initializing distributed domain in initializer

### ISSUE THREAD

**Issue by benharsh** [chapel-lang/chapel#6488]
### Summary of Problem
The following code fails to compile:

```chpl
use BlockDist;

class Foo {
  var dom : domain(3);
  var space = dom dmapped Block(dom);
  var A : [space] int;

  proc init(n) {
    dom = {1..n,1..n,1..n};
    super.init();
  }
}

proc main() {
  var f = new Foo(10);
}
```

The compiler emits the following error:
```
foo.chpl:8: In initializer:
foo.chpl:5: error: unresolved type specifier 'Block(domain(3,int(64),false))'
$CHPL_HOME/modules/dists/BlockDist.chpl:304: note: candidates are: Block(param rank: int(64), type idxType, type sparseLayoutType)
```

I am working off of master, and this program fails as of a7f0d2b5a18a61763ed7c7bfe793af458ed2a21e

**Comment by benharsh** [comment]
@lydia-duncan : here's the issue we discussed.

**Comment by benharsh** [comment]
I believe this was resolved by @noakesmichael in #6567
